lib/schema_plus/core/sql_struct.rb

```diff
@@ -12,6 +16,7 @@ def parse!(sql)
             \( \s*
             (?<body>.*) \s*
             \) \s*
+            #{INHERITANCE_REGEX if sql[INHERITANCE_KEY]}
```

Comment:
I see yes you're right.  In that case my main comment is that `sql[INHERITANCE_KEY]` could match something unexpected in a non-inherited table's definition (unlikely but possible) , so i'd suggest using `sql.match INHERITANCE_REGEX` (even less likely to match unexpectedly).  And still I'd go with fewer constants.  Plus it could use a comment to explain what's going on. 

Here's what I came up with:  (While in there, I realized that `^` and `$` should really be `\A` and `\Z`)

```ruby
        INHERITANCE_REGEX = %r{ \s* (?<inheritance>INHERITS \s* \( [^)]* \)) }mxi

        def parse!(sql)
          m = sql.strip.match %r{
          \A
          (?<command>.*\bTABLE\b) \s*
            (?<quote>['"`])(?<name>\S+)\k<quote> \s*
            \( \s*
            (?<body>.*) \s*
            \) \s*
            # can't use optional ? for inheritance because it would be greedily grabbed into body;
            # ideally this would use an actual parser rather than regex
            #{INHERITANCE_REGEX if sql.match INHERITANCE_REGEX}
            (?<options> \S.*)?
          \Z
          }mxi
          self.command = m[:command]
          self.quotechar = m[:quote]
          self.name = m[:name]
          self.body = m[:body]
          self.options = m[:options]
          self.inheritance = m[:inheritance] rescue nil
        end
````